upload/common/upload_api_client.py

```diff
@@ -0,0 +1,30 @@
+from urllib.parse import urljoin
```

Comment:
Is this really "common" code or is the api_client worthy of its own folder under upload/?  Not sure.